Mask generation for training dataset

navjot-nangia opened the pull request:
*** working on it ***

These scripts uses:
- pretrained detection model for weed detection
- unet or sam for initial mask generation
- post segmentation script for refining mask based on either hsv filtreing, EXG thresholding etc.

navjot-nangia committed:
Enhanced voxel_cutouts.py to support optional prediction masks

navjot-nangia committed:
Merge branch 'add_voxe51_script' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into add_voxe51_script

navjot-nangia committed:
minor doc improvement

navjot-nangia committed:
Add weed detection, segmentation and post-segmentation processing scripts for mask generation

- Created ProcessDetections class to handle batch processing of images and saving results.
- UNET or SAM segmentation inference using a pre-trained UNet model or SAM-HQ.
- Added draw_bbox.py for interactive bounding box drawing and YOLO format saving if needed
- Introduced post-segment processing with ExG thresholding and mask refinement with HSV color ranges in post_segment_processing.py.

navjot-nangia committed:
sam script for initial mask generation

navjot-nangia committed:
useful utils for mask generation

navjot-nangia committed:
improved documenation

navjot-nangia committed:
Add script to copy selected images from long-term storage to destination directory

navjot-nangia committed:
Added hydra conf to run scripts with main.py.

navjot-nangia committed:
modified detect_weeds to run with cfg

navjot-nangia committed:
Update docstring for weed detection script with detailed usage and configuration requirements

navjot-nangia committed:
Refactor UNet segmentation script: enhance docstring, update script to use Hydra config

navjot-nangia committed:
Refactor post-segmentation processing: update class name, enhance logging, and streamline image processing methods

mkutu committed:
updated directory structure for mask_gen

mkutu committed:
overview image

mkutu committed:
minor reorg

navjot-nangia committed:
Enhance configuration and inspection scripts: add mask inspection paths, update dataset handling, and added FiftyOneMaskInspector class for improved image tagging workflow.

navjot-nangia committed:
added config file for morphological operation in mask_gen

navjot-nangia committed:
added destination directory for mask to be moved into selected_dataset_for_training_dir

navjot-nangia committed:
deleted unnecessary scripts

navjot-nangia committed:
Refactor mask inspection script: enhance documentation, improve dataset handling, and add functionality for moving tagged images to long-term storage.

navjot-nangia committed:
Added functinality to move images/masks tagged 'good' to long-term storage, improve path handling, and update documentation.

navjot-nangia committed:
Implement mask refinement module: add RefineMask class for processing and refining segmentation masks using HSV thresholding and morphological operations.

navjot-nangia committed:
minor

navjot-nangia committed:
made a better mask_refine config file

navjot-nangia committed:
updated paths

navjot-nangia committed:
moved detect weeds and unet segmentation to main pipeline; other minor moving

mkutu commented:
* [x] implement `project` structure for `mask_gen` outputs
* [x] implement testing for `mask_gen` pipeline
    - uploading 2-3 smaller images and unet masks to github placing them in a test folder that represent 1-2 mask issues (missing red, missing white)
    - do not save to lts during testing
* [x] update main config to include:
    1. mode and tasks
    2. save2lts flag for mask_inspect
    3. test.enabled
* [x] separate mask issue refinement methods in mask_refine.py into separate refinement scripts

mkutu committed:
mkutu corrections

navjot-nangia committed:
moved missing color and present mat functionality to mask_gen_utils

navjot-nangia committed:
refactor: updated configuration and logging for mode handling in mask inspection.

mkutu committed:
- setup primary and secondary entry points
- adjusted configs to use mode/task
- added/udpated diagrams
- included directory structure md for projects format

mkutu committed:
added test data (testing scripts)

mkutu commented:
refer to #45 for updates

navjot-nangia committed:
minor fix: update output filenames to include suffix for clarity

navjot-nangia committed:
added environemnt.yaml file version 1.0

navjot-nangia committed:
update configuration paths, added functionality to skip no detection bbox and updated terminal instruction docs for voxel

navjot-nangia committed:
edited dir structure to avoid repeated directory name (i.e: projec.name)

navjot-nangia committed:
updated how project name is being used in paths and changed the name from csv to db

navjot-nangia committed:
removed unnecessary directory creation

navjot-nangia committed:
refactor voxel inspection paths and added another column to database

navjot-nangia committed:
- improved process_cuout_dir by refactoring fuctionality into separate functions for readability
- handled bad, other, and unknown voxel tags
- updates docs

navjot-nangia committed:
improved _handle_tagged_image method

navjot-nangia committed:
removes initial and refined masks for bad and other tags

navjot-nangia committed:
added the deleted test images

mkutu commented on mask_generation/environment.yaml:
remove "version" and add specific package version

mkutu committed:
directory structure and environment update

mkutu committed:
added test data

mkutu committed:
mask_gen utils reorg and cleaning

navjot-nangia committed:
- corrected project name in config
- made sure the correct project name and sub-direcotries are used in config paths

mkutu committed:
ignore updates to test project folder
